Tests/Functional/ViewHelpers/PeriodRegistrationViewHelperTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package jweiland/reserve.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace JWeiland\Reserve\Tests\Functional\ViewHelpers;
+
+use JWeiland\Reserve\Domain\Repository\PeriodRepository;
+use JWeiland\Reserve\Service\ReserveService;
+use Nimut\TestingFramework\TestCase\FunctionalTestCase;
+use Prophecy\Argument;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Object\ObjectManager;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+
+class PeriodRegistrationViewHelperTest extends FunctionalTestCase
+{
+    protected $testExtensionsToLoad = ['typo3conf/ext/reserve'];
+
+    /**
+     * @var StandaloneView
+     */
+    protected $standaloneView;
+
+    /**
+     * @var \DateTime
+     */
+    protected $testDateMidnight;
+
+    /**
+     * @var \DateTime
+     */
+    protected $testDateAndBegin;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->standaloneView = GeneralUtility::makeInstance(ObjectManager::class)->get(StandaloneView::class);
+
+        $this->importDataSet(__DIR__ . '/../Fixtures/example_facility_with_period.xml');
+
+        $this->testDateMidnight = new \DateTime('+2 days midnight');
+        $this->testDateAndBegin = clone $this->testDateMidnight;
+        $this->testDateAndBegin->setTime(14, 00);
+
+        GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getConnectionForTable('tx_reserve_domain_model_period')
+            ->update(
+                'tx_reserve_domain_model_period',
+                ['date' => $this->testDateMidnight->getTimestamp()],
+                ['deleted' => 0]
+            );
+
+        GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getConnectionForTable('tx_reserve_domain_model_period')
+            ->update(
+                'tx_reserve_domain_model_period',
+                ['begin' => (new \DateTime('1970-01-01T14:00:00.00Z'))->getTimestamp()],
+                ['uid' => 1]
+            );
+
+        $this->setTestTemplate(1, $this->testDateAndBegin);
+    }
+
+    protected function setTestTemplate(int $facilityUid, \DateTime $dateAndBegin): void
+    {
+        $this->standaloneView->setTemplateSource(
```

Comment:
IMO it's not your task to test a complete template. Just test the code in the ViewHelper and you're done.

Reply:
My first thought was like calling `renderStatic()` and am fine but then I thought it would be more realistic to really "use" the ViewHelper in a dummy template and render it to get the result.

Reply:
In that case I prefer using real template files like done here: https://github.com/jweiland-net/events2/blob/master/Resources/Private/Templates/Day/ListThisWeek.txt

Reply:
I could add an example template that uses this ViewHelper in the Tests Folder or like `Resources/Private/Templates/Test/ViewHelper.html` or so because I don't use the ViewHelper inside the extension templates.